ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/server/execution/ConnectExecutorTest.java

```diff
@@ -118,13 +128,31 @@ public void shouldReturnConnectorInfoEntityOnSuccess() {
   }
 
   @Test
-  public void shouldReturnErrorEntityOnError() {
+  public void shouldReturnErrorEntityOnCreationError() {
     // Given:
-    givenError();
+    givenValidationSuccess();
+    givenCreationError();
 
     // When:
     final Optional<KsqlEntity> entity = ConnectExecutor
-        .execute(CREATE_CONNECTOR_CONFIGURED, mock(SessionProperties.class), null, serviceContext).getEntity();
+        .execute(CREATE_CONNECTOR_CONFIGURED, mock(SessionProperties.class), null,
+            serviceContext).getEntity();
+
+    // Then:
+    assertThat("Expected non-empty response", entity.isPresent());
+    assertThat(entity.get(), instanceOf(ErrorEntity.class));
+  }
+
+  @Test
+  public void shouldReturnErrorEntityOnValidationError() {
+    // Given:
+    givenValidationError();
+    givenCreationSuccess();
+
+    // When:
+    final Optional<KsqlEntity> entity = ConnectExecutor
+        .execute(CREATE_CONNECTOR_CONFIGURED, mock(SessionProperties.class), null,
+            serviceContext).getEntity();
 
     // Then:
     assertThat("Expected non-empty response", entity.isPresent());
```

Comment:
Is it difficult to add a check for the actual error message returned to the user in this test? It'd be nice to see an example of the error message produced to ensure that the formatting is sane, and to prevent accidental regressions in the future. (Looks fine from the code you added but it's always nice to double-check in a test if possible.)

Reply:
Added in 77f6de69656c41aac1f463d2c993e797dc1ba4f0.